iceoryx2-ffi/cxx/include/iox2/listener.hpp

```diff
@@ -83,6 +83,8 @@ class Listener {
   private:
     template <ServiceType>
     friend class PortFactoryListener;
+    template <ServiceType>
+    friend class WaitSet;
```

Comment:
Just curious, what is this required for?

Reply:
The `WaitSet` actually attaches only file descriptors and this allows the `WaitSet` to call `iox2_listener_get_file_descriptor` with the `m_handle`.

Reply:
Yes, the user could a lot of stupid things with it and it would restrict us - then every listener would need to be file descriptor based and we have building blocks that are based on a semaphore. This would not work with this API.